src/main/java/core/gui/comp/table/HOTableModel.kt

```diff
@@ -0,0 +1,402 @@
+        }
+        return table.columnModel.getColumn(i)
+    }
+
+    /**
+     * Store user table settings in the database if they were changed by the user
+     */
+    fun storeUserSettings() {
+        if (table == null) return
+        var changed = setUserColumnSettings(table!!)
+
+        val sorter = table!!.rowSorter as RowSorter<HOTableModel>
+        if (sorter != null) {
+            if (setRowOrderSettings(sorter)) {
+                changed = true
+            }
+        }
+        if (changed) {
+            DBManager.instance().saveHOColumnModel(this)
+        }
+    }
+
+    /**
+     * Get row order from user columns and restore it to the given row sorter
+     *
+     * @param rowSorter Row sorter
+     */
+    private fun getRowOrderSettings(rowSorter: RowSorter<HOTableModel>) {
+        // Restore row order setting
+        val sortKeys = ArrayList<RowSorter.SortKey>()
+        val sortColumns = Arrays.stream(this.columns).filter { i: UserColumn -> i.sortPriority != null }.sorted(
+            Comparator.comparingInt { obj: UserColumn -> obj.getSortPriority() }).toList()
+        if (!sortColumns.isEmpty()) {
+            val userColumns = Arrays.stream(this.columns).toList()
+            for (col in sortColumns) {
+                val index = userColumns.indexOf(col)
+                val sortKey = RowSorter.SortKey(index, col.getSortOrder())
+                sortKeys.add(sortKey)
+            }
+        }
+        rowSorter.sortKeys = sortKeys
+    }
+
+    /**
+     * Set user columns sort priority and order from given row sorter
+     * @param sorter Row sorter
+     * @return True if one user setting is changed
+     * False, if no user settings are changed
+     */
+    private fun setRowOrderSettings(sorter: RowSorter<HOTableModel>): Boolean {
+        var changed = false
+        val rowSortKeys = sorter.sortKeys
+        for (i in columns.indices) {
+            val finalI = i
+            val rowSortKey = rowSortKeys.stream().filter { k: RowSorter.SortKey -> k.column == finalI }.findFirst()
+            val userColumn = columns[i]
+            if (rowSortKey.isPresent && rowSortKey.get().sortOrder != SortOrder.UNSORTED) {
+                val k = rowSortKey.get()
+                val priority = rowSortKeys.indexOf(k)
+                if (userColumn.getSortPriority() == null || userColumn.getSortPriority() != priority || userColumn.getSortOrder() != k.sortOrder) {
+                    userColumn.setSortOrder(k.sortOrder)
+                    userColumn.setSortPriority(priority)
+                    changed = true
+                }
+            } else if (userColumn.getSortPriority() != null) {
+                userColumn.setSortPriority(null)
+                userColumn.setSortOrder(null)
+                changed = true
+            }
+        }
+        return changed
+    }
+
+    companion object {
+        @Serial
+        private val serialVersionUID = -207230110294902139L
+    }
```

Comment:
I am still not sure whether there is any point to `serialVersionUID`.  My understanding was that this was used in serialization/deserialization, and to ensure that the changes made to a serializable class were compatible so that older version of a serialized instance can still be deserialized (I remember a loooong time this mattered for example in RMI), but I don't believe we make sure changes are compatible, or if serialization is even used here.